### ISSUE TITLE
How to use this benchmark to evaluate other models, such as SDXL and SD3-medium?

### ISSUE THREAD

**Issue by DthdZK** [Karine-Huang/T2I-CompBench#18]

**Comment by Karine-Huang** [comment]
To use this benchmark to evaluate other models, such as SDXL and SD3-medium, follow these steps:

1. Generate Images:

- Generate 10 images for each prompt across all categories.
- You can refer to the [inference_eval.py](https://github.com/Karine-Huang/T2I-CompBench/blob/main/GORS_finetune/inference_eval.py) script to generate images, fix the seed, and save them in the formatted image files.
Example directory structure for saving the images:

```
color/samples/
    ├── a green bench and a blue bowl_000000.png
    ├── a green bench and a blue bowl_000001.png
    └──...
```
2. Evaluation:

- Refer to the "Evaluation" section under "Example usage" in the Readme to evaluate the model's performance across different categories.
- Use the following tools for evaluation:
  - BLIP-VQA for attribute binding (color, shape, texture).
  - Unidet for spatial relationships (2D and 3D) and numeracy.
  - CLIPScore for non-spatial relationships.
  - 3-in-1 for complex compositions.
  - MLLM evaluation for all categories (optional).

I hope this helps! Let me know if you need further assistance.

**Comment by YuehengLuo** [comment]
Hi, I would like to know how to evaluate various metrics if I want to use a generative model like sd1.5 should I use color_val.txt to generate 3000 images and then use bash BLIPvqa_eval/test.sh to get a score that is Attribute color? And then the test Attribute Shape has to be generated using Shape_val.txt？ I mean when I want to reproduce the corresponding metrics, I should use the corresponding val.txt to generate the test image, right?

**Comment by Karine-Huang** [comment]
Yes, you are correct. To evaluate various metrics for a generative models, use the corresponding val.txt files to generate the test images for each category.

**Comment by YuehengLuo** [comment]
Thank you. I get it！

**Comment by sayakpaul** [comment]
@Karine-Huang would be nice to include this piece of information in the README to make the order of evaluation more explicit. In the current form, it appears like we need to first fine-tune and then evaluate which is probably not the case.

**Comment by Karine-Huang** [comment]
Thanks for your advice, that change would be clearer for evaluation!

Also, "spatial.txt" should contain the contents of both "spatial_train.txt" and "spatial_val.txt". We have fixed that! Thank you for pointing that out!

**Comment by mrwu-mac** [comment]
> To use this benchmark to evaluate other models, such as SDXL and SD3-medium, follow these steps:
> 
> 1. Generate Images:
> 
> * Generate 10 images for each prompt across all categories.
> * You can refer to the [inference_eval.py](https://github.com/Karine-Huang/T2I-CompBench/blob/main/GORS_finetune/inference_eval.py) script to generate images, fix the seed, and save them in the formatted image files.
>   Example directory structure for saving the images:
> 
> ```
> color/samples/
>     ├── a green bench and a blue bowl_000000.png
>     ├── a green bench and a blue bowl_000001.png
>     └──...
> ```
> 
> 2. Evaluation:
> 
> * Refer to the "Evaluation" section under "Example usage" in the Readme to evaluate the model's performance across different categories.
> * Use the following tools for evaluation:
>   
>   * BLIP-VQA for attribute binding (color, shape, texture).
>   * Unidet for spatial relationships (2D and 3D) and numeracy.
>   * CLIPScore for non-spatial relationships.
>   * 3-in-1 for complex compositions.
>   * MLLM evaluation for all categories (optional).
> 
> I hope this helps! Let me know if you need further assistance.

@Karine-Huang Hi, after fixing the seed, why do we still need to generate 10 images for each prompt? Shouldn't these 10 images be the same under a fixed seed? Looking forward to your reply.

**Comment by mrwu-mac** [comment]
> > To use this benchmark to evaluate other models, such as SDXL and SD3-medium, follow these steps:
> > 
> > 1. Generate Images:
> > 
> > 
> > * Generate 10 images for each prompt across all categories.
> > * You can refer to the [inference_eval.py](https://github.com/Karine-Huang/T2I-CompBench/blob/main/GORS_finetune/inference_eval.py) script to generate images, fix the seed, and save them in the formatted image files.
> >   Example directory structure for saving the images:
> > 
> > ```
> > color/samples/
> >     ├── a green bench and a blue bowl_000000.png
> >     ├── a green bench and a blue bowl_000001.png
> >     └──...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 2. Evaluation:
> > 
> > 
> > * Refer to the "Evaluation" section under "Example usage" in the Readme to evaluate the model's performance across different categories.
> > * Use the following tools for evaluation:
> >   
> >   * BLIP-VQA for attribute binding (color, shape, texture).
> >   * Unidet for spatial relationships (2D and 3D) and numeracy.
> >   * CLIPScore for non-spatial relationships.
> >   * 3-in-1 for complex compositions.
> >   * MLLM evaluation for all categories (optional).
> > 
> > I hope this helps! Let me know if you need further assistance.
> 
> [@Karine-Huang](https://github.com/Karine-Huang) Hi, after fixing the seed, why do we still need to generate 10 images for each prompt? Shouldn't these 10 images be the same under a fixed seed? Looking forward to your reply.

After carefully reviewing `inference_eval.py`, I now understand your point. I apologize for any inconvenience caused.

**Comment by Karine-Huang** [comment]
> > > To use this benchmark to evaluate other models, such as SDXL and SD3-medium, follow these steps:
> > > 
> > > 1. Generate Images:
> > > 
> > > 
> > > * Generate 10 images for each prompt across all categories.
> > > * You can refer to the [inference_eval.py](https://github.com/Karine-Huang/T2I-CompBench/blob/main/GORS_finetune/inference_eval.py) script to generate images, fix the seed, and save them in the formatted image files.
> > >   Example directory structure for saving the images:
> > > 
> > > ```
> > > color/samples/
> > >     ├── a green bench and a blue bowl_000000.png
> > >     ├── a green bench and a blue bowl_000001.png
> > >     └──...
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > 2. Evaluation:
> > > 
> > > 
> > > * Refer to the "Evaluation" section under "Example usage" in the Readme to evaluate the model's performance across different categories.
> > > * Use the following tools for evaluation:
> > >   
> > >   * BLIP-VQA for attribute binding (color, shape, texture).
> > >   * Unidet for spatial relationships (2D and 3D) and numeracy.
> > >   * CLIPScore for non-spatial relationships.
> > >   * 3-in-1 for complex compositions.
> > >   * MLLM evaluation for all categories (optional).
> > > 
> > > I hope this helps! Let me know if you need further assistance.
> > 
> > 
> > [@Karine-Huang](https://github.com/Karine-Huang) Hi, after fixing the seed, why do we still need to generate 10 images for each prompt? Shouldn't these 10 images be the same under a fixed seed? Looking forward to your reply.
> 
> After carefully reviewing `inference_eval.py`, I now understand your point. I apologize for any inconvenience caused.

Sorry about the confusion in the statement. The seed is set here [inference](https://github.com/Karine-Huang/T2I-CompBench/blob/main/GORS_finetune/inference_eval.py), line 102 & 106.
